Comorphisms/test/casl_features.casl

```diff
@@ -0,0 +1,91 @@
+spec test_if =
+  pred p,q : ()
+  . p if q
+end
+
+spec test_overloading =
+  sorts s < t1; t1 < u; s < t2; t2 < u
+  sorts v < w; w < x
+  ops f:s->v; f:s->x; f:t1->w  %% in the overloading relation
+  ops g:s->v; g:s->x; g:t1->w  %% in the overloading relation
+  forall y:s
+  . f(y)=g(y) %(f=g)%
+end
+
+spec test_non_overloading =
+  sorts s < t1; t1 < u; s < t2; t2 < u
+  sorts v < w; w < x
+  sorts z1,z2
+  ops f:s->v; f:s->z1; f:z1->w; f:z2->w  %% not in the overloading relation
+  ops g:s->v; g:s->z1; g:z1->w; g:z2->w  %% not in the overloading relation
+  forall y:s
+  . (op f : s -> v)(y) = (op g : s -> v)(y) %(f=g 1)%
+  . (op f : s -> z1)(y) = (op g : s -> z1)(y) %(f=g 2)%
+  forall y:z1
+  . f(y) = g(y) %(f=g 3)%
+end
+
+spec test_non_overloading2 =
+  sorts s < t1; t1 < u; s < t2; t2 < u
+  sorts v < w; w < x
+  sorts z1,z2
+  ops f:s->v; f:t1->z1; f:v->w; f:w->t1  %% not in the overloading relation
+  ops g:s->v; g:t1->z1; g:v->w; g:w->t1  %% not in the overloading relation
+end
+
+spec test_partially_overloading =
+  sorts s < t1; t1 < u; s < t2; t2 < u
+  sorts v < w; w < x
+  sort z
+  ops f:s->v; f:s->x; f:z->w  %% partially in the overloading relation
```

Comment:
I suggest to change this into `ops f:s->v; f:s->x; f:s->z`. Then, the comorphism currently returns three different operations: `op_f_s_v`, `op_f_s_x`, `op_f_s_z`, but it should return only two. The former two operations should be combined into one, because `ops f:s->v; f:s->x` are in the overloading relation.

Reply:
When exactly does "being in the overloading relation" mean? Is it that all of these are true?
* the op name is the same
* the arguments have the same type
* the types of the results are in the same connected component

Reply:
it is a bit more complicated, see p.27 bottom of http://www.informatik.uni-bremen.de/cofi/CASL-RM.pdf

Reply:
Is there already a function somewhere in Hets that checks if two functions are in the overloading relation?

Reply:
`CASL.Overload.leqF` and (for predicates) `CASL.Overload.leqP`

Reply:
Should the former two be combined into this? 
```tptp
fof(sign_op_1_f_s_x, axiom,
    ! [VAR_X1]: ((sort_s(VAR_X1)) => sort_x(op_f_s_x(VAR_X1)))).
```

Reply:
This is the axiom for just one operations. As I said, there should be two operations (and such an axiom for each of them).
When thinking about it: perhaps it is easier to re-use the comorphism CASL2TopSort here. (Let us discuss this face to face.)

Reply:
I'm currently working on it and can distinguish the two cases. I'm just not 100% sure what the result should be.

Should this:
```casl
spec test_partially_overloading =
  sort s
  sorts v < w; w < x
  sort z
  ops f:s->v; f:s->x; f:s->z  %% partially in the overloading relation
end
```
be translated to these axioms for the ops?
```tptp
fof(sign_op_1_f_s_x, axiom,
    ! [VAR_X1]: ((sort_s(VAR_X1)) => sort_x(op_f_s_x(VAR_X1)))).
fof(sign_op_f_s_z, axiom,
    ! [VAR_X1]: ((sort_s(VAR_X1)) => sort_z(op_f_s_z(VAR_X1)))).
```

Reply:
yes

Reply:
I think I have fixed it in 711e0cad1. Please verify.

Reply:
Note (also for myself): I used
```
hets -v2 --translation=CASL2TPTP_FOF -n test_partially_overloading Comorphisms/test/casl_features.casl -o tptp -O /tmp
```
to run the translation